src/librustc_hir/def.rs

```diff
@@ -235,6 +275,7 @@ pub enum Namespace {
     TypeNS,
     ValueNS,
     MacroNS,
+    LifetimeNS,
```

Comment:
cc @petrochenkov For this `Namespace` addition.
I guess long-term if we start resolving named lifetimes in `rustc_resolve` (which I even started a branch for ages ago, I should finish that...), we'll need something like this anyway.

Reply:
I'm skeptical about this.
Yes, lifetimes live in a separate namespace, but unlike the three namespaces listed in `Namespace` they are syntactically different so we never need to ask the question "is this `ns` a `LifetimeNS`"?
See how labels are treated in `rustc_resolve`, which are also in a separate namespace (different from lifetimes), but don't have a variant in `Namespace`.

Reply:
> which are also in a separate namespace (different from lifetimes), but don't have a variant in `Namespace`

Makes sense, and you could probably say `Namespace` is only for names that paths can refer to.

Reply:
@petrochenkov @eddyb Ok, I will remove it. Should I also remove it from `PerNs`?

Reply:
>Should I also remove it from `PerNs`?

Why? It's used a lot in resolve.